src/smart-components/group/validators.js

```diff
@@ -1,25 +1,24 @@
 import { fetchGroups } from '../../helpers/group/group-helper';
 import asyncDebounce from '../../utilities/async-debounce';
 
-export const asyncValidator = async (groupName) => {
-  if (!groupName) {
-    return undefined;
-  }
-
-  if (groupName.length > 150) {
-    throw 'Can have maximum of 150 characters.';
-  }
+const validationPromise = (name, idKey, id) =>
+  name.length < 150
+    ? fetchGroups({ limit: 10, offset: 0, name, nameMatch: 'exact' }).then(({ data }) => {
+        if (data.length === 0) {
+          return undefined;
+        }
 
-  const response = await fetchGroups({ limit: 10, offset: 0, name: groupName, nameMatch: 'exact' }).catch((error) => {
-    console.error(error);
-    return undefined;
-  });
+        const taken = data.some((item) => item[idKey] !== id && item.name === name);
```

Comment:
Here you are kinda duplicating logic that is done by API when you pass `nameMatch: "exact"`

Reply:
yeah, thanks, siplified it